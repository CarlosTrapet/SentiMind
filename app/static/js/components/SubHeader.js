import React from 'react';

const SubHeader = () => {
  return (
    <div>
      <h3>Type in a topic, find out what the twitterverse thinks</h3>
    </div>
  );
}

export default SubHeader;
