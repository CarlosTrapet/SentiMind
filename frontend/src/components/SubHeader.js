import React from 'react';
import StyledSubHeader from './styled_components/SubHeader';

const SubHeader = props => {
  return (
    <div>
      <StyledSubHeader>
        <p id="subheader">{props.subHeaderText}</p>
      </StyledSubHeader>
    </div>
  );
};

export default SubHeader;
