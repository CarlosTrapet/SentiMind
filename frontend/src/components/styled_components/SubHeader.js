import styled from 'styled-components';

const StyledSubHeader = styled.h2`
  color: white;
  font-size: 36px;
`;

export default StyledSubHeader;
