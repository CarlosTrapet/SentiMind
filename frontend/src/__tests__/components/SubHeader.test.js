import React from 'react';
import { shallow } from 'enzyme';
import SubHeader from '../../components/SubHeader';

describe('SubHeader', () => {
  const props = {text: 'Flexible Subheader'}
  const subHeader = shallow(<SubHeader {...props}/>)

  it('renders correctly', () => {
    expect(subHeader).toMatchSnapshot()
  })

  it('renders flexible subheader', () => {
    expect(subHeader.find('h3').text()).toEqual('Flexible Subheader')
  })
})
