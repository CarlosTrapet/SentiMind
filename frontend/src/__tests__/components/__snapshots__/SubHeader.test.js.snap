// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubHeader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h3>
        Type in a topic, find out what the twitterverse thinks
</h3>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Type in a topic, find out what the twitterverse thinks",
      },
      "ref": null,
      "rendered": "Type in a topic, find out what the twitterverse thinks",
      "type": "h3",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h3>
          Type in a topic, find out what the twitterverse thinks
</h3>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Type in a topic, find out what the twitterverse thinks",
        },
        "ref": null,
        "rendered": "Type in a topic, find out what the twitterverse thinks",
        "type": "h3",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
